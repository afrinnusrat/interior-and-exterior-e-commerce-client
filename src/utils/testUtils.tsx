import React from 'react';
import { BrowserRouter, Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../modules/store';
import { createMemoryHistory } from 'history';

const history = createMemoryHistory();

export const withRouter = (Component: any) => (props: any) => (
  <BrowserRouter>
    <Component {...props} />
  </BrowserRouter>
);

export const withHistory = (Component: any) => (props: any) => (
  <Router history={history}>
    <Component {...props} />
  </Router>
);

export const withProvider = (Component: any) => (props: any) => (
  <Provider store={store}>
    <BrowserRouter>
      <Component {...props} />
    </BrowserRouter>
  </Provider>
);
