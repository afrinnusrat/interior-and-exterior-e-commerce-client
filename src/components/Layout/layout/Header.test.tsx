import { render, cleanup, fireEvent } from '@testing-library/react';
import { Header, HeaderProps } from './Header.component';
import { withProvider } from '../../../utils/testUtils';

beforeEach(cleanup);

const HeaderWithProvider = withProvider(Header);

describe('<Header />', () => {
  const mockHandleOpen = jest.fn();
  const mockToggleSearchForm = jest.fn();

  const props: HeaderProps = {
    handleOpen: mockHandleOpen,
    toggleSearchForm: mockToggleSearchForm
  };

  const setup = (props: HeaderProps) => {
    const utils = render(HeaderWithProvider(props));
    return { ...utils };
  };

  it('renders correctly', () => {
    const { asFragment } = setup(props);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls `handleOpen` func', () => {
    const { getByTestId } = setup(props);
    fireEvent.click(getByTestId('Hamburger'));

    expect(mockHandleOpen).toHaveBeenCalledTimes(1);
  });

  it('calls `toggleSearchForm` func', () => {
    const { getByTestId } = setup(props);
    fireEvent.click(getByTestId('SearchBtn'));

    expect(mockToggleSearchForm).toHaveBeenCalledTimes(1);
  });
});
