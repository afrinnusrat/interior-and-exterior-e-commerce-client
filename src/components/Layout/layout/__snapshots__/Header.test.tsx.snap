// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<DocumentFragment>
  <header
    class="sc-htoDjs dHOETM"
  >
    <div
      class="sc-dnqmqq gTOaEr"
    >
      <a
        class="sc-iwsKbI hQBmBE"
        href="/"
      >
        <h1
          class="sc-gZMcBi fPQmCU"
        >
          Interior & Exterior
        </h1>
      </a>
      <nav
        class="sc-gqjmRU eqrLjR"
      >
        <button
          aria-label="open menu"
          class="sc-VigVT fwtpGC"
          data-testid="Hamburger"
          title="Menu"
        >
          <i
            class="fas fa-bars"
          />
        </button>
        <ul
          class="sc-bdVaJa hDTYr"
        >
          <li
            class="sc-bxivhb icElMN"
          >
            <a
              class="sc-htpNat gyqNkk"
              data-testid="PageLink"
              href="/shop"
            >
              Products
              <i
                class="fas fa-caret-down"
              />
            </a>
            <div
              class="sc-bwzfXH gjnxxD"
            >
              <ul
                class="sc-ifAKCX kqqTha"
              >
                <li
                  class="sc-EHOje jTwYBT"
                >
                  <a
                    class="sc-bZQynM imRHDl"
                    data-testid="Chairs"
                    href="/shop"
                  >
                    Chairs
                  </a>
                </li>
                <li
                  class="sc-EHOje jTwYBT"
                >
                  <a
                    class="sc-bZQynM imRHDl"
                    data-testid="Lamps"
                    href="/shop"
                  >
                    Lamps
                  </a>
                </li>
                <li
                  class="sc-EHOje jTwYBT"
                >
                  <a
                    class="sc-bZQynM imRHDl"
                    data-testid="Clocks"
                    href="/shop"
                  >
                    Clocks
                  </a>
                </li>
                <li
                  class="sc-EHOje jTwYBT"
                >
                  <a
                    class="sc-bZQynM imRHDl"
                    data-testid="Vases"
                    href="/shop"
                  >
                    Vases
                  </a>
                </li>
                <li
                  class="sc-EHOje jTwYBT"
                >
                  <a
                    class="sc-bZQynM imRHDl"
                    data-testid="Benches"
                    href="/shop"
                  >
                    Benches
                  </a>
                </li>
                <li
                  class="sc-EHOje jTwYBT"
                >
                  <a
                    class="sc-bZQynM imRHDl"
                    data-testid="Flowerpots"
                    href="/shop"
                  >
                    Flowerpots
                  </a>
                </li>
                <li
                  class="sc-EHOje jTwYBT"
                >
                  <a
                    class="sc-bZQynM imRHDl"
                    data-testid="Plants"
                    href="/shop"
                  >
                    Plants
                  </a>
                </li>
              </ul>
            </div>
          </li>
          <li
            class="sc-bxivhb icElMN"
          >
            <a
              class="sc-htpNat gyqNkk"
              href="/account"
            >
              Account
            </a>
          </li>
          <li
            class="sc-bxivhb icElMN"
          >
            <a
              class="sc-htpNat gyqNkk"
              href="/cart"
            >
              Cart (0)
            </a>
          </li>
        </ul>
        <button
          class="sc-jTzLTM kYnXAK"
          data-testid="SearchBtn"
        >
          <i
            class="fas fa-search"
          />
        </button>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
